test/Constraints/closures.swift

```diff
@@ -182,8 +182,7 @@ func r22162441(_ lines: [String]) {
 
 func testMap() {
   let a = 42
-  [1,a].map { $0 + 1.0 } // expected-error {{binary operator '+' cannot be applied to operands of type 'Any' and 'Double'}}
-  // expected-note @-1 {{expected an argument list of type '(Double, Double)'}}
+  [1,a].map { $0 + 1.0 } // expected-error {{cannot convert value of type 'Int' to expected element type 'Double'}}
```

Comment:
It's still not quite right is it? The diagnostic talks about an 'element type' but that's Int here isn't it?

Reply:
It's not really `Int`, since there is a literal which could be a `Double` as well, which would match `+` inside of the closure body.

Reply:
I see it from the perspective that if there was no `a` in there, everything would have type-checked correctly as `[Double]`.

Reply:
Only other possible failure we could produce would say that `$0` expected to be `Double` which might not be more useful since it still comes down to use of `a`.